packages/lesswrong/lib/collections/posts/views.ts

```diff
@@ -816,13 +816,12 @@ ensureIndex(Posts,
 );
 
 Posts.addView("onlineEvents", (terms: PostsViewTerms) => {
-  const yesterday = moment().subtract(1, 'days').toDate();
   let query = {
     selector: {
       onlineEvent: true,
       isEvent: true,
       groupId: null,
-      $or: [{startTime: {$exists: false}}, {startTime: {$gt: yesterday}}],
+      $or: [{startTime: {$gt: new Date()}}, {endTime: {$gt: new Date()}}],
```

Comment:
IMO this should show events with a start time but no end time, and should show recently-ended events. I have a pretty common experience on Facebook, where someone will create an event for a party and not specify an end time or leave a default 1h duration, and I go looking for it 2h after the start time to get the address so I can show up fashionably late.

Reply:
Asking a related question in #joint, will address though. Note that this *does* allow for events with no end time.

Reply:
I think my solution is correct here.

Reply:
I made this change explicitly because of negative reactions from my user interviews - users were confused/annoyed to see the events listed at the **top** of the Community home page be past events that were no longer relevant. I think this downside is pretty bad, since it makes this page significantly less useful if you have to individually check events to make sure they are still happening now/in the future.

I think events on the EA Forum tend to be more like lectures/discussion groups (where it's not reasonable to show up 2 hours late) and less like parties. So maybe we can just split this logic by forum type. Or maybe we make start and end times required so that this is less of an issue?

Reply:
Are you modeling that, because we were comparing the event time to the server time, we were cutting off all events <8 hours in the future in California?

Reply:
Perhaps I'm missing something? `startTime` is stored in UTC right? And the server has a consistent timezone as well, so when we query for posts with `startTime` greater than now on the server, timezones should not affect the result.

When we show event times we don't always show the timezone, so maybe that's why you thought we were incorrectly excluding some future events.

For example, I made this event in a group based in Berkeley (so the event inherits the location of Berkeley, even though it's an online event). The event starts at 1pm PT but since I'm in ET it shows 4pm instead. But you might be assuming that the event starts at 4pm PT since the group is based in Berkeley, so you might be surprised that it's no longer listed on the Community home page after 4pm ET, because you might expect it to be there until 7pm ET.
<img width="354" alt="Screen Shot 2021-11-03 at 2 57 51 PM" src="https://user-images.githubusercontent.com/9057804/140175201-3271f28a-4716-4e46-9c94-ad30616446b8.png">
The database has `startTime: ISODate("2021-11-03T20:00:00.000Z")`, which is 1pm PT / 4pm ET.

Definitely possible that I'm missing something because...timezones

Reply:
Alright, I'm just completely mis-modeling how events' times are stored. I really though I remembered that we weren't converting their time to UTC, and I thought I asked about that in slack, but I think I misinterpreted Jim's answer. I'm trying to address comments on like 10 different PRs, so it was a bit hard to dig in deeply to everything, but I should have been less confident here.

I've given it another go. The EA Forum is very conservative, the only delta from @s-cheng's work is an hour buffer on the start time. For LW I've tried to strike a balance between the user feedback that I believe would still apply to LW, and Jim's feedback about his facebook experience. Unless Jim's showing up fashionably 3-hours-after-end-time-late, I think he'll be satisfied.